### PR TITLE
test: change some skipped tests to use retries

### PR DESCRIPTION
**Pull request by flotwig**
- Closes 

### Additional details

- Followup to #23167 and #23386
- Instead of skipping, sets retries to 15 on tests marked as flaky in those PRs
- Some consistent failures exist now, where previously there was flake:
	1. `uses cy.origin twice` in `origin/validation.cy.ts` https://dashboard.cypress.io/projects/ypt4pf/runs/40003/test-results/0c4b2400-e352-49e2-90f6-fdc91506aa9e
	2. `xhr log has response body/status code when xhr response is logged second` in `proxy_logging.cy.ts` https://dashboard.cypress.io/projects/ypt4pf/runs/40003/test-results/7d0e5b2a-c299-4ebb-8478-fdfbfcb39ef2 
	3. `xhr log has response body/status code when xhr response is logged first` in `proxy_logging.cy.ts` https://dashboard.cypress.io/projects/ypt4pf/runs/40003/test-results/d4c78b51-b525-462d-a844-a4803d263799
		- seems to pass rarely
	4. `verifies fetch requests made while a secondary origin is active eventually update with snapshots of the secondary origin` in `origin/snapshots.cy.ts`
		- seems to pass rarely
	5. `sends Studio usage metadata` in `record_spec` https://app.circleci.com/pipelines/github/cypress-io/cypress/44422/workflows/25bf1f35-9741-4f30-b29c-6ec11cf7e89c/jobs/1861142
	6. `passes in parallel with group` in `record_spec` https://app.circleci.com/pipelines/github/cypress-io/cypress/44422/workflows/25bf1f35-9741-4f30-b29c-6ec11cf7e89c/jobs/1861142
	7. `makes the right command for pnpm` / `makes the right command for npm` in `project-setup.cy.ts` https://app.circleci.com/pipelines/github/cypress-io/cypress/45902/workflows/14823615-1e9c-48fd-8d1b-b75a314110dc/jobs/1932073
	8. `saves the filter when navigating to a spec and back` in `specs_list_e2e.cy.ts` https://app.circleci.com/pipelines/github/cypress-io/cypress/45902/workflows/14823615-1e9c-48fd-8d1b-b75a314110dc/jobs/1932099
	9. `passes filtered options to Project#open and sets cli config` in `cypress_spec.js` https://app.circleci.com/pipelines/github/cypress-io/cypress/46741/workflows/a44414d3-e3ab-4614-aeb3-5c041a87400f/jobs/1960772
	10. `takes the user to first step of ct setup when switching from app` in `project-setup.cy.ts` https://app.circleci.com/pipelines/github/cypress-io/cypress/46741/workflows/a44414d3-e3ab-4614-aeb3-5c041a87400f/jobs/1960739
	11. `shows no icon and file system timestamp for files` in `specs_list_no_git_repo.cy.ts` https://app.circleci.com/pipelines/github/cypress-io/cypress/47574/workflows/c287ce0b-245b-449e-9bab-86a66f0265dd/jobs/1990772

### PR Tasks

- [x] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/network/test/integration/connect_spec.ts [hunk 1]
```diff
@@ -11,8 +11,8 @@ chai.use(sinonChai)
 
 describe('lib/connect', function () {
   context('.getAddress', function () {
-    // TODO: fix flaky test https://github.com/cypress-io/cypress/issues/23155
```
Comment (mjhenkes) [inline packages/network/test/integration/connect_spec.ts:14]: Should we keep the TODO here?

File: packages/server/lib/browsers/webkit.ts [hunk 2]
```diff
@@ -121,7 +121,7 @@ export async function open (browser: Browser, url: string, options: BrowserLaunc
      */
     private suppressUnhandledEconnreset () {
       unhandledExceptions.handle((err: NodeJS.ErrnoException) => {
-        return err.code === 'ECONNRESET'
+        return err.code !== 'ECONNRESET'
```
Comment (mjhenkes) [inline packages/server/lib/browsers/webkit.ts]: What is this doing here? Should this have it's own issue to track it?

File: packages/network/test/integration/connect_spec.ts [hunk 3]
```diff
@@ -11,8 +11,8 @@ chai.use(sinonChai)
...
-    it.skip('resolves localhost on 127.0.0.1 immediately', function () {
+    it('resolves localhost on 127.0.0.1 immediately', function () {
+      this.retries(15)
```
Comment (mjhenkes) [inline packages/network/test/integration/connect_spec.ts:16]:
```suggestion
      // TODO: fix flaky test https://github.com/cypress-io/cypress/issues/23155
      this.retries(15)
```

File: packages/server/lib/browsers/webkit.ts (hunk 2 above)
Comment (flotwig) [reply packages/server/lib/browsers/webkit.ts]: Mistakenly included this in the diff, I'll open a separate PR for it.

```suggestion
        return err.code === 'ECONNRESET'
```

File: packages/network/test/integration/connect_spec.ts (hunk 1 above)
Comment (flotwig) [reply packages/network/test/integration/connect_spec.ts:14]: Ya

**Review by mjhenkes — APPROVED** [review]

**Comment by rachelruderman** [comment]
Thank you for implementing this!!! Definitely preferable to boost retries where possible than to skip altogether. I see some CI is failing, is that expected? Ideally the changes would keep CI green (unless there's unrelated CI failures happening)

**Comment by flotwig** [comment]
> Ideally the changes would keep CI green (unless there's unrelated CI failures happening)

@rachelruderman it seems to be flake, but not flake that was touched by these changes

**Review by rachelruderman — APPROVED** [review]
Sounds good!! Thank you for taking this on 🙏

**Comment by flotwig** [comment]
Had to re-skip some of the retries, there's a full list in the OP comment. I think this is now at a good spot where it's not going to bring any flake/failures back.

**Pull request merged**